snorkel/models/candidate.py

```diff
@@ -99,8 +99,8 @@ def candidate_subclass(class_name, args, table_name=None, cardinality=None,
             raise ValueError("Number of values must match cardinality.")
         if None in values:
             raise ValueError("`None` is a protected value.")
-        if any([type(v) == int for v in values]):
-            raise ValueError("Values cannot be integers.")
+        if any([type(v) == int and v == 0 for v in values]):
```

Comment:
So the intended usage was as follows: if you want integer-valued classes (e.g. `[1,2,3]`), you just leave `values` unset and then it defaults to that.

So technically, the only reason to specify integer values is if you want to specify a non-consecutive sequence of ints... but this will not be correctly handled in the current [code](https://github.com/HazyResearch/snorkel/blob/master/snorkel/annotations.py#L412). Overall, this is a minor issue; but either way options would be to:
* Just revert back to the previous code, but add a note to the error message saying to just leave `values` unset
* Check for non-consecutive or zero-valued ints, throw error then